src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,80 @@
+package duke;
+
+/**
+ * Represent the class that deals with interactions with the user
+ * CS2103T
+ * AY23/24 Semester 2
+ * Author: Chua Zen Khoon
+ */
+public class Ui {
+
+    private Parser parser;
+    public Ui(){
+        parser = new Parser();
+    }
+
+
+    public void showLoadingError(DukeException e) {
+        System.out.println(e.getMessage());
+    }
+
+    /**
+     * Analyses the input of the user and parses out the relevant details
+     * for the correct action to be taken in the main Duke class
+     *
+     * @param echo input to be analysed
+     * @return an integer array with the correct action and number in the TaskList
+     *          to operate on
+     */
+    public int[] analyseUserInput(String echo) {
+        int[] results = new int[] {0,0};
+
+        if (echo.equals("bye") || echo.equals("Bye")) {
+            results[0] = 1;
+        } else if (echo.equals("list") || echo.equals("List")) {
```

Comment:
can use echo.toLowerCase() method to convert all the letters to lowercase

Reply:
Thanks for the suggestions and review!